dashboard/app/views/levels/editors/_blockly.html.haml

```diff
@@ -183,99 +183,111 @@
       matchBrackets: true
     });
 
-    var callouts = callout_editor.getValue();
+    var callout_spaces = [];
+    var updated_callouts = [];
+    var space_id = 0;
+
+    fill_form_with_existing_callouts();
 
-    if (callouts.length > 0) {
-      var parsecallouts = JSON.parse(callouts);
+    function fill_form_with_existing_callouts() {
+      var callout_json = callout_editor.getValue();
+      var callouts = [];
+
+      if(callout_json.length > 0) {
+        callouts = JSON.parse(callout_json);
+      }
 
       for(var i=0; i<callouts.length; i++) {
-        if (parsecallouts[i]) {
-          $(".use_callout_" + i).prop("checked", true);
-          $(".localization_key_" + i).val(parsecallouts[i].localization_key);
-          $(".callout_text_" + i).val(parsecallouts[i].callout_text);
-          $(".element_id_" + i).val(parsecallouts[i].element_id);
-          $(".on_" + i).val(parsecallouts[i].on);
-          if ((parsecallouts[i].qtip_config) && (parsecallouts[i].qtip_config.position))
-            $(".qtip_position_my_" + i).val(parsecallouts[i].qtip_config.position.my);
-          if ((parsecallouts[i].qtip_config) && (parsecallouts[i].qtip_config.position))
-            $(".qtip_position_at_" + i).val(parsecallouts[i].qtip_config.position.at);
-          if ((parsecallouts[i].qtip_config) && (parsecallouts[i].qtip_config.position) && (parsecallouts[i].qtip_config.position.adjust))
-            $(".qtip_position_adjust_x_" + i).val(parsecallouts[i].qtip_config.position.adjust.x);
-          if ((parsecallouts[i].qtip_config) && (parsecallouts[i].qtip_config.position) && (parsecallouts[i].qtip_config.position.adjust))
-            $(".qtip_position_adjust_y_" + i).val(parsecallouts[i].qtip_config.position.adjust.y);
-          if ((parsecallouts[i].qtip_config) && (parsecallouts[i].qtip_config.style) && (parsecallouts[i].qtip_config.style.classes) && (parsecallouts[i].qtip_config.style.classes == "no-tip"))
-            $(".qtip_hide_" + i).prop("checked", true);
+        if(callouts[i].callout_text) {
+          create_new_callout_space();
+          id = callout_spaces[i];
+          $(".callout_" + id + " .localization_key").val(callouts[i].localization_key);
```

Comment:
Nit: if you want you can set `var calloutSpace = $('.callout_' + i);` above and then do `calloutSpace.find('.localization_key')` to search within the current calloutSpace.